From 1e2cf1f5229bc6ca80afaa5aaede5a6234204bef Mon Sep 17 00:00:00 2001
From: ZhongWen <b47623@freescale.com>
Date: Fri, 27 Sep 2013 14:29:26 +0800
Subject: [PATCH 026/115] ENGR00278171 wfd-sink: update wfd-sink apk

        Do some optimization for the apk:
        1.add the WIFI check function.
        2.add the fullscreen function.
        3.modify the refresh button.

Signed-off-by: ZhongWen <b47623@freescale.com>
---
 WfdSink/res/layout/sink_main.xml                   | 86 ++++++++++----------
 WfdSink/res/layout/sink_pic.xml                    |  1 +
 WfdSink/res/values/strings.xml                     |  9 ++-
 WfdSink/res/values/styles.xml                      |  2 +-
 .../src/com/freescale/wfdsink/WfdSinkActivity.java | 91 +++++++++++-----------
 5 files changed, 97 insertions(+), 92 deletions(-)

diff --git a/WfdSink/res/layout/sink_main.xml b/WfdSink/res/layout/sink_main.xml
index b9f622b..8408d92 100755
--- a/WfdSink/res/layout/sink_main.xml
+++ b/WfdSink/res/layout/sink_main.xml
@@ -17,8 +17,8 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@drawable/bg"
 	android:layout_gravity="center"
+	android:background="#000000"
     android:orientation="vertical" >
 
     <RelativeLayout
@@ -41,15 +41,7 @@
             <RelativeLayout
                 style="@style/style_quicksettings_category_title"
                 android:layout_width="match_parent"
-                android:layout_height="wrap_content" >
-
-                <TextView
-                    style="@style/style_quicksettings_category_text"
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:layout_alignParentLeft="true"
-                    android:layout_centerVertical="true"
-                    android:text="@string/search" />
+                android:layout_height="match_parent" >
 
                 <ImageButton
                     android:id="@+id/sink_player"
@@ -58,9 +50,50 @@
                     android:layout_alignParentRight="true"
                     android:layout_centerVertical="true"
                     android:layout_marginRight="30dp"
-                    android:src="@drawable/ic_refresh" />
+                    android:background="@drawable/ic_refresh" />
+
+                <ImageView
+                    android:id="@+id/imageView1"
+                    android:layout_width="60dp"
+                    android:layout_height="60dp"
+                    android:layout_alignParentLeft="true"
+                    android:layout_centerVertical="true"
+                    android:layout_marginLeft="17dp"
+                    android:src="@drawable/ic_hdmi" />
+
+                <LinearLayout
+                    android:id="@+id/linearLayout2"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_centerVertical="true"
+                    android:layout_marginLeft="253dp"
+                    android:layout_toRightOf="@+id/imageView1"
+                    android:orientation="vertical" >
+                </LinearLayout>
+
+                <TextView
+                    android:id="@+id/currentdevice"
+                    style="@style/style_quicksettings_category_text"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_alignTop="@+id/linearLayout2"
+                    android:layout_marginLeft="16dp"
+                    android:layout_toRightOf="@+id/imageView1"
+                    android:text="@string/current"
+                    android:textSize="25sp" />
+
+                <TextView
+                    android:id="@+id/status"
+                    style="@style/style_quicksettings_category_text"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_alignBottom="@+id/imageView1"
+                    android:layout_alignLeft="@+id/currentdevice"
+                    android:text="@string/status"
+                    android:textSize="25sp" />
 
             </RelativeLayout>
+
         </LinearLayout>
 
         <GridView
@@ -72,36 +105,5 @@
             android:numColumns="4" >
         </GridView>
 
-        <LinearLayout
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:layout_alignParentBottom="true"
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentRight="true"
-            android:layout_below="@+id/gridview"
-            android:orientation="vertical" >
-
-            <TextView
-                android:id="@+id/text"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:text="@string/current"
-                android:textSize="30sp" />
-
-            <ImageView
-                android:id="@+id/imageView1"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:src="@drawable/ic_hdmi" />
-
-            <TextView
-                android:id="@+id/currentdevice"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:text="@string/current"
-                android:textSize="25sp" />
-
-        </LinearLayout>
-
     </RelativeLayout>
 </LinearLayout>
diff --git a/WfdSink/res/layout/sink_pic.xml b/WfdSink/res/layout/sink_pic.xml
index 1d0dbf2..5ccaa02 100755
--- a/WfdSink/res/layout/sink_pic.xml
+++ b/WfdSink/res/layout/sink_pic.xml
@@ -18,5 +18,6 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_gravity="center"
+        android:textColor="#ffe0e0e0"
         android:gravity="center_horizontal" />
 </LinearLayout>
diff --git a/WfdSink/res/values/strings.xml b/WfdSink/res/values/strings.xml
index a6ffb03..eee7812 100755
--- a/WfdSink/res/values/strings.xml
+++ b/WfdSink/res/values/strings.xml
@@ -5,7 +5,10 @@
     <string name="hello_world">Hello world!</string>
     <string name="menu_settings">Settings</string>
     <string name="title_activity_second">SecondActivity</string>
-	<string name="search">Avaiable devices</string>
-	<string name="current">Current device</string>
-
+    <string name="search">Avaiable devices</string>
+    <string name="current">Local device Information</string>
+    <string name="status">status</string>
+    <string name="setting_key">setting_key</string>
+    <string name="setting_summary">setting_summary</string>
+    <string name="setting_title">setting_title</string>	
 </resources>
diff --git a/WfdSink/res/values/styles.xml b/WfdSink/res/values/styles.xml
index e431dfd..f3ea816 100755
--- a/WfdSink/res/values/styles.xml
+++ b/WfdSink/res/values/styles.xml
@@ -28,4 +28,4 @@
         <item name="android:padding">4dp</item>
     </style>	
 
-</resources>
\ No newline at end of file
+</resources>
diff --git a/WfdSink/src/com/freescale/wfdsink/WfdSinkActivity.java b/WfdSink/src/com/freescale/wfdsink/WfdSinkActivity.java
index c6cc756..e2d615d 100755
--- a/WfdSink/src/com/freescale/wfdsink/WfdSinkActivity.java
+++ b/WfdSink/src/com/freescale/wfdsink/WfdSinkActivity.java
@@ -25,7 +25,7 @@ import android.view.View;
 import android.view.GestureDetector;
 import android.widget.VideoView;
 import android.view.MotionEvent;
-
+import android.net.wifi.WifiManager;
 import android.util.Log;
 import android.media.MediaPlayer;
 import android.media.MediaPlayer.OnCompletionListener;
@@ -44,9 +44,12 @@ import java.util.TimerTask;
 import java.util.List;
 import java.util.ArrayList;
 import android.net.Uri;
+import android.content.Intent;
+import android.provider.Settings;
 import android.widget.ImageButton;
 import android.widget.ImageView;
 import android.widget.ArrayAdapter;
+import android.widget.LinearLayout;
 import android.widget.Spinner;
 import android.view.Display;
 import android.widget.AdapterView.OnItemSelectedListener;
@@ -102,8 +105,9 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
     private boolean mButtonShow = false;
     private Timer mTimer;
     private String mThisName;
-	private TextView currentdevice;
-
+    private TextView currentdevice;
+    private TextView status;
+    private LinearLayout layout;    
     private final BroadcastReceiver mWifiP2pReceiver = new BroadcastReceiver() {
         public void onReceive(Context context, Intent intent) {
             String action = intent.getAction();
@@ -141,7 +145,7 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
                 currentdevice = (TextView)findViewById(R.id.currentdevice);
                 if (mThisName != mWfdSink.getDeviceName())
                     mThisName = mWfdSink.getDeviceName();
-                currentdevice.setText(mWfdSink.getDeviceName());
+                currentdevice.setText("SSID:" + mWfdSink.getDeviceName());
             }
 
         }
@@ -167,10 +171,17 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
         SurfaceHolder holder = mSurfaceView.getHolder();
         holder.addCallback(this);
         holder.setType(SurfaceHolder.SURFACE_TYPE_PUSH_BUFFERS);
-
+	    WifiManager manager = (WifiManager) getSystemService(WIFI_SERVICE);
         mGridView = (GridView)findViewById(R.id.gridview);
         mImageButton = (ImageButton)findViewById(R.id.sink_player);
 
+        status = (TextView)findViewById(R.id.status);
+        layout = (LinearLayout)findViewById(R.id.linearLayout1);
+	if(manager.isWifiEnabled()){
+				
+		}else{
+		     wifiDialog();
+		     }		
         mImageButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View arg0) {
@@ -184,31 +195,11 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
             }
         });
 
-        mGridView.setOnTouchListener(new View.OnTouchListener() {
-            public boolean onTouch(View v, MotionEvent mv) {
-            switch (v.getId()) {
-            case R.id.gridview:
-            Log.i(TAG, "onClick");
-            if (!mStarted && !mConnected) {
-            mImageButton.setVisibility(View.VISIBLE);
-            mTimer.schedule(new TimerTask() {
-                public void run () {
-                mHandler.sendEmptyMessage(UPDATE_BUTTON_SHOW);
-                }
-                }, 4000);
-            }
-            break;
-            }
-                return false;
-            }
-        });
-
         mPictureAdapter = new PictureAdapter(this);
         mDisplayMetrics = new DisplayMetrics();
         getWindowManager().getDefaultDisplay().getMetrics(mDisplayMetrics);
 
         mSurfaceView.setVisibility(View.INVISIBLE);
-        mImageButton.setVisibility(View.GONE);
         mGridView.setVisibility(View.VISIBLE);
         mGridView.setAdapter(mPictureAdapter);
 
@@ -225,6 +216,29 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
             return super.onKeyDown(keyCode, event);
         }
 
+	private void wifiDialog()
+	{
+		Dialog dialog = new AlertDialog.Builder(WfdSinkActivity.this)
+		.setTitle("WI-FI hasn't open ! ").setMessage("Are you sure to open now? If you cancel then will exit!").setIcon(R.drawable.ic_hdmi)
+		.setPositiveButton("confirm", new DialogInterface.OnClickListener() {
+			
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				
+				startActivity(new Intent(Settings.ACTION_WIFI_SETTINGS));
+
+			}
+		}).setNegativeButton("cancel", new DialogInterface.OnClickListener() {
+			
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				System.exit(-1);
+				
+			}
+		}).create();
+		dialog.show();
+	}		
+		
     private void exitDialog() {
         Dialog dialog = new AlertDialog.Builder(WfdSinkActivity.this)
             .setTitle("Program exit ?")
@@ -257,25 +271,22 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
                     mGridView.setAdapter(mPictureAdapter);
                     break;
 
-                case UPDATE_BUTTON_SHOW:
-                    mButtonShow = false;
-                    mImageButton.setVisibility(View.GONE);
-                    break;
-
                 case UPDATE_SURFACE:
                     SurfaceHolder holder = (SurfaceHolder)msg.obj;
                     handleUpdateSurface(holder);
                     break;
 
-                case START_PLAY:
+                case START_PLAY:                	
                     handleStartPlay();
                     break;
 
                 case STOP_PLAY:
+                	status.setText("Status:Disconnected!");
                     handleStopPlay();
                     break;
 
                 case DO_CONNECTED:
+                	status.setText("Status:Connected");
                     boolean connected = (msg.arg1 == 1);
                     handleConnected(connected);
                     break;
@@ -332,9 +343,8 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
         mPictureAdapter.setSourcePeers(mSourcePeers);
         mGridView.setAdapter(mPictureAdapter);
         mGridView.postInvalidate();
-
+        layout.setVisibility(View.VISIBLE);
         mSurfaceView.setVisibility(View.INVISIBLE);
-        mImageButton.setVisibility(View.GONE);
         mGridView.setVisibility(View.VISIBLE);
         mStarted = false;
     }
@@ -342,6 +352,7 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
     private void handleConnected(boolean connected) {
         mConnected = connected;
         if (mConnected) {
+        	layout.setVisibility(View.INVISIBLE);
             mSurfaceView.setVisibility(View.VISIBLE);
             mGridView.setVisibility(View.GONE);
             startPlayer();
@@ -352,6 +363,7 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
     }
 
     private void startSearch() {
+    	status.setText("Status:Searching for peers");
         mWfdSink.startSearch();
         //mWfdSink.setDeviceName("Android_me");
     }
@@ -378,19 +390,6 @@ public class WfdSinkActivity extends Activity implements SurfaceHolder.Callback
         mHandler.sendMessage(msg);
     }
 
-    public boolean onTouchEvent(MotionEvent event) {
-        super.onTouchEvent(event);
-        Log.w(TAG, "onTouchEvent");
-        if (!mStarted && !mConnected) {
-            mImageButton.setVisibility(View.VISIBLE);
-            mTimer.schedule(new TimerTask() {
-                public void run () {
-                    mHandler.sendEmptyMessage(UPDATE_BUTTON_SHOW);
-                }
-            }, 4000);
-        }
-        return true;
-    }
 
     public void onStart() {
         super.onStart();
-- 
1.8.0

